src/zed_cpu_ros.cpp

```diff
@@ -370,15 +373,36 @@ class ZedCameraROS {
 	 * @param      img_pub       The image pub
 	 * @param[in]  img_frame_id  The image frame identifier
 	 * @param[in]  t             { parameter_description }
+	 * @param[in]  encoding      image_transport encoding
 	 */
-	void publishImage(cv::Mat img, image_transport::Publisher &img_pub, std::string img_frame_id, ros::Time t) {
+	void publishImage(cv::Mat img, image_transport::Publisher &img_pub, std::string img_frame_id, ros::Time t, const std::string& encoding) {
```

Comment:
Can you try change the input to const cv::Mat& img?
And test if it works with a subscriber.